### DISCUSSION TITLE
Any recent alterations in the integration that could affect battery charge rate

### DISCUSSION THREAD

**Discussion by richsimpson** [wills106/homeassistant-solax-modbus#255] [Q&A] [answered]
Over the past few updates here in the UK it has been pretty cool weather.

During that time I see sometimes my storage batteries (3 x Solax 5.8's) will charge at different rates. Sometimes they will get from 11% to 100% in the 4 hour charging period I have set for cheap energy, other times they will only reach 80%.

Just wondering if there is anything that is throttling the inverter or battery or if just down to the temperatures? They appear to start at a lower rate of charge then speed up, previously stepping up to 3.4kwh quite quickly, last night only as high as that after 2.5 hrs of charging and a few step up in the charge rate.

last night was warmer and this:
![image](https://user-images.githubusercontent.com/51749826/208297722-a3bf1ff4-0dc6-4fa3-b978-63143bcbb041.png)

Earlier in the week when much cooler was charging at more around 3.4kwh through the majority of the period. My HASS only stores 3 days of history but from the solax app:

last night
![image](https://user-images.githubusercontent.com/51749826/208297869-459d5baa-517b-4aef-9fdc-445c037bdc26.png)

Earlier in the week a higher charge rate (days where it was similarly dull and so charging from empty to full)
![image](https://user-images.githubusercontent.com/51749826/208297909-99e9ace2-6a53-4e9c-aef6-8036453db25b.png)

![image](https://user-images.githubusercontent.com/51749826/208297941-f4e4ae6d-4f6e-4369-be6f-7cec4b8bf23f.png)

the inverter (Solax x1 g4) is in Manual Mode and Force Charge during 2030-0030hrs taking advantage of Octopus Go Faster tariff. The house is wholly powered from the grid during that time, car charging and other stuff going to pulling all power from grid. inverter is only putting power into battery, I would have expected it to be doing that at its maximum 3.6kwh which is perhaps the 3.4kwh that I see at the peak. But why not throughout, particularly when its doing the 11%-80% part of the battery?

**Comment by wills106** [comment, marked as answer]
I am guessing it's purely down to temps.

I have 3x4.5kWh the ramup last night was over a 10min period. My batteries were 20oC and peaked at 24oC though.

![image](https://user-images.githubusercontent.com/18155231/208299334-7ea01415-a2b4-42ea-a780-a104136701ba.png)

**Comment by richsimpson** [comment]
Further investigation and the BMS Charge Max Current is what is creating the steps, so presumably the inverters protection over the battery packs based on temperature (as you suggested) and/or something else. Its a much warmer day but still seeing the steps.

Its stepping from 7 to 8 to 9 Amps. Maybe beyond too. 

Just forced charge and watched the sensor in the integration.

Inverter 33'c and batteries 16'c it was at 8.0A and actual current to batteries 8.3A.
Stepped up to 9.0A Max Current from the BMS and the batteries actually seeing 9.4A Temps both stayed the same at that point and beyond. 

Will keep an eye on it and now I have that sensor enabled I have a bit more info to review through the charging phase now. More than the Solax website for sure!

**Comment by richsimpson** [comment]
For info, last nights warmer weather did indeed see the battery charge at higher rates and although it hadn't quite got down to the 11% minimum SOC, it completed the charge from 14% to 98% in the 4 hour window with 17 mins to spare in fact. 

I set up a small chart to monitor overnight
![image](https://user-images.githubusercontent.com/51749826/208647211-5745e3d5-24b2-4a6e-961c-2ebdf47d33e8.png)
BMS Max Charge rate _(red line)_ did indeed step up in increments. Peaking at 25A when nearly fully charged although the actual charge rate is limited by the inverter so didn't go beyond 9.8A.

My 'issue' therefore is when the battery charges below 9A

Last night that didn't happen, most likely due to the higher temperature, either of the battery or the ambient temperature in the garage. 

![image](https://user-images.githubusercontent.com/51749826/208648385-bd9b1c4a-68b7-4682-b662-b95fd796e0c8.png)

Strangely the BMS Max Charge rate fluctuates through the early hours whilst the battery is discharging (from 00:30hrs onwards). House load relatively flat and the temperature slowly falling a couple of 'c.

**Reply by wills106** [reply]
From reading on a few Forums and FB pages this isn't scientific...
But there seems to be two temps to take into consideration:
8oC Batteries slow down charging like you experienced and 2oC you effectively have a paper weight.

Did you notice what temps your batteries dropped to when you had the issue?

Looks like the T3.0 Batteries have built in heaters, but if I was to go for new again I would take your T5.8 Batteries over them as I am borderline on capacity with my older 4.5 ones.

Also with your T5.8 you can use that fancy external BMS and double your capacity.

I know the T3.0 battery heaters can be controlled through this Integration if I implement it, but I don't know if you have any ports on your batteries to wire one? Or perhaps on that external BMS?

**Reply by richsimpson** [reply]
> Did you notice what temps your batteries dropped to when you had the issue?
> 
> Looks like the T3.0 Batteries have built in heaters, but if I was to go for new again I would take your T5.8 Batteries over them as I am borderline on capacity with my older 4.5 ones.
> 
> Also with your T5.8 you can use that fancy external BMS and double your capacity.
> 
> I know the T3.0 battery heaters can be controlled through this Integration if I implement it, but I don't know if you have any ports on your batteries to wire one? Or perhaps on that external BMS?

Not sure on the temps when it was really slow but it would have been low. During the recent cold spell. Not as low as 2'c but perhaps 8'c. They are in the garage although the batteries temp is always higher than ambient in there. 

Last night it was a fairly flat line of maximum flow. 9 to 10A flow, with the BMS increasing through the charge although my inverter is maxed out at that point so I don't see any benefit from the max going beyond 10A anyway. 
![image](https://user-images.githubusercontent.com/51749826/209316419-13792802-8531-4bd8-9b8c-a819ad5a6906.png)
 
And all the rest of that info is interesting but went a bit over my head. No external BMS for me, presume some option. The other option I didn't take up was the emergency supply switch over. I know I can add stuff to the output to handle but I must say I was a bit miffed that in a power-cut I also go dark. Even with my 16KWH of power sitting doing nothing. Different story for a different day though. Only really relevant if we were to start to get blackouts

**Comment by MikeShipp** [comment]
This is the SolaX charge data for various temperatures and charge levels
![IMG_0328](https://user-images.githubusercontent.com/6832143/213249563-c3e58838-710a-44b2-8491-07ec56afa1f8.jpeg)

**Reply by wills106** [reply]
Which battery is this for?

Ignoring non SolaX ones I think there are 3 main groups.

T45 / T6 (4.5 / 6kWh)

T3 (3kWh with Battery Heater)

T58 (5.8kWh no Battery Heater)

Edit:
Looks like they are also Introducing a new Pylontech style setup as well. (Possibly similar to what you could use on the Gen2 Hybrids?)

**Reply by richsimpson** [reply]
> This is the SolaX charge data for various temperatures and charge levels ![IMG_0328](https://user-images.githubusercontent.com/6832143/213249563-c3e58838-710a-44b2-8491-07ec56afa1f8.jpeg)

Interesting but not what I experience. Where is this info from for the Triple Power T58 battery?

Right now its 15'c on the battery and the max charge rate is 7A. Capacity currently 66% so by that chart I should be seeing more like 25A. Perhaps the battery could but doesn't because of the firmware that Solax have written? 

![image](https://user-images.githubusercontent.com/51749826/213410092-d0d43365-aefe-4765-bb72-c0288111c6ab.png)

**Comment by MikeShipp** [comment]
Triple Power 5.8kWh (50Ah). As you say, unfortunately no battery heater.

I’m not currently using your integration but using my own RESTful (no modbus) sensors in YAML. I’ve played with modbus using a TCP client and it communicates with my X1-Fit-G4 inverter to read Holding Registers and Input Registers. I was looking at just using standard modbus sensors in YAML but your work looks interesting. I did buy an RS485 adapter (USB) but I’ve managed to communicate using modbus over TCP on the pocket WiFi (not local mode) and wondered why your integration suggests that this is not satisfactory.

I’m currently trying to get SolaX to upgrade my firmware versions, which may kill my current setup. However, once the firmware is up to date I’ll have a go with your integration.

I’m new to Home Assistant and SolaX so don’t necessarily understand how it all works!

On 18 Jan 2023, at 17:27, wills106 ***@***.***> wrote:

﻿

Which battery is this for?

Ignoring non SolaX ones I think there are 3 main groups.

T45 / T6 (4.5 / 6kWh)

T3 (3kWh with Battery Heater)

T58 (5.8kWh no Battery Heater)

**Reply by wills106** [reply]
> modbus over TCP on the pocket WiFi (not local mode) and wondered why your integration suggests that this is not satisfactory.

Pocket WiFi falls under three categories:
- It work's
- It doesn't work
- It works when it want's too...

The V3 looks more promising at working though?

It doesn't help when people on FB claim Pocket WiFi works with this Integration 100%.

Similar to people on FB with Gen3 Inverters advising Gen4 owners to just plug their Inverter into networking equipment.

Just because a Gen4 has RJ45 sockets, doesn't mean it supports ethernet over RJ45!

So it's easier for me to say Pocket WiFi is not supported. I don't own a Pocket WiFi and I don't know which variation / Firmware people are using. So it's easier to say it's not supported.

**Comment by MikeShipp** [comment]
I understand. I’ll give it a go before switching to RS485 (if it doesn’t work).

Mike

On 18 Jan 2023, at 18:38, wills106 ***@***.***> wrote:

﻿

modbus over TCP on the pocket WiFi (not local mode) and wondered why your integration suggests that this is not satisfactory.

Pocket WiFi falls under three categories:

  *   It work's
  *   It doesn't work
  *   It works when it want's too...

The V3 looks more promising at working though?

It doesn't help when people on FB claim Pocket WiFi works with this Integration 100%.

Similar to people on FB with Gen3 Inverters advising Gen4 owners to just plug their Inverter into networking equipment.

Just because a Gen4 has RJ45 sockets, doesn't mean it supports ethernet over RJ45!

So it's easier for me to say Pocket WiFi is not supported. I don't own a Pocket WiFi and I don't know which variation / Firmware people are using. So it's easier to say it's not supported.

**Reply by wills106** [reply]
Good luck and I'm glad you understand :)

**Comment by MikeShipp** [comment]
I also found that it didn't really behave as the chart suggests according to the battery temperature but did behave according to the ambient temperature. I did query with SolaX where they measure the temperature but didn't really get an answer. You would think that battery temperature would be the correct option. I do have a tubular heater which I used to prove the point but as long as they charge in 4 hours I don't really mind.
  
​

**Comment by alienatedsec** [comment]
Each battery is managed by the internal BMS and will set its charging current based on its internal protections. That is based on many factors, including temp (reducing the current to warm up the battery first or keep it low if the battery temp is not increasing while charging), SoC (reducing the current when the battery is almost full - e.g, around 95%).

I found that some battery brands won't charge if outside temps are around -10C for a longer period. Looking at the spec, they will stop discharging when -20C.
